axiom/util_strings.h

```diff
@@ -471,4 +473,35 @@ static inline bool nr_striendswith(const char* s,
   return 0 == nr_stricmp(suffix, pattern);
 }
 
+/*
+ * Purpose : Strip the ".php" file extension from a file name
+ *
+ * Params  : 1. The string filename
+ *           2. The filename length
+ *
+ * Returns : A newly allocated string stripped of the .php extension
+ *
+ */
+static inline char* nr_file_basename(char* filename, int filename_len) {
+  char* retval = NULL;
+
+  if (NULL == filename || 0 >= filename_len) {
+    return NULL;
+  }
+
+  if (4 >= filename_len) {
+    /* if filename_len <= 4, there can't be a ".php" substring to remove. Assume
+     * the filename does not contain ".php" and return the original filename. */
+    return filename;
+  }
```

Comment:
There really is no need for this check - this is the first thing `nr_striendswith` does and `nr_striendswith` is inline.

Reply:
#789 addresses this